core/src/main/java/io/questdb/cutlass/line/tcp/DelegatingTlsChannel.java

```diff
@@ -0,0 +1,435 @@
+                KeyStore jks = KeyStore.getInstance("JKS");
+                try (InputStream trustStoreStream = openTruststoreStream(trustStorePath)) {
+                    jks.load(trustStoreStream, trustStorePassword);
+                }
+                tmf.init(jks);
+                TrustManager[] trustManagers = tmf.getTrustManagers();
+                sslContext.init(null, trustManagers, new SecureRandom());
+            } else if (validationMode == Sender.TlsValidationMode.INSECURE) {
+                sslContext = SSLContext.getInstance("TLS");
+                sslContext.init(null, BLIND_TRUST_MANAGERS, new SecureRandom());
+            } else {
+                sslContext = SSLContext.getDefault();
+            }
+
+            // SSLEngine needs to know hostname during TLS handshake to validate a server certificate was issued
+            // for the server we are connecting to. For details see the comment below.
+            // Hostname validation does not use port at all hence we can get away with a dummy value -1
+            SSLEngine sslEngine = sslContext.createSSLEngine(expectedHostname.toString(), -1);
+            if (validationMode != Sender.TlsValidationMode.INSECURE) {
+                SSLParameters sslParameters = sslEngine.getSSLParameters();
+                // The https validation algorithm? That looks confusing! After all we are not using any
+                // https here at so what does it mean?
+                // It's actually simple: It just instructs the SSLEngine to perform the same hostname validation
+                // as it does during HTTPS connections. SSLEngine does not do hostname validation by default. Without
+                // this option SSLEngine would happily accept any certificate as long as it's signed by a trusted CA.
+                // This option will make sure certificates are accepted only if they were issued for the
+                // server we are connecting to.
+                sslParameters.setEndpointIdentificationAlgorithm("https");
+                sslEngine.setSSLParameters(sslParameters);
+            }
+            sslEngine.setUseClientMode(true);
+            return sslEngine;
+        } catch (Throwable t) {
+            if (t instanceof LineSenderException) {
+                throw (LineSenderException)t;
+            }
+            throw new LineSenderException("error while creating openssl engine", t);
+        }
+    }
+
+    private static InputStream openTruststoreStream(String trustStorePath) throws FileNotFoundException {
+        InputStream trustStoreStream;
+        if (trustStorePath.startsWith("classpath:")) {
+            String adjustedPath = trustStorePath.substring("classpath:".length());
+            trustStoreStream = DelegatingTlsChannel.class.getResourceAsStream(adjustedPath);
+            if (trustStoreStream == null) {
+                throw new LineSenderException("Configured trust at classpath:" + trustStorePath + " is unavailable on a classpath");
```

Comment:
this is quite wordy, we also try to follow exception semantics, e.g

```
factual statement [contextVar1=value, contextVar2=value...]
```

```suggestion
                throw new LineSenderException("could not load trust store from classpath [path=" + trustStorePath + "]");
```

lets cross-check with other clients how they offer user to supply certificates. Working with classpath from library is quite tricky and should be avoided.

Reply:
I would offer user to supply stream, which we can close for them if they like

Reply:
Why is it tricky? imagine you are embedding a QuestDB client in e.g. Spring app. 

Embedding a trust store in your application UberJAR is the simplest thing an application developer can do. Just place it into `/src/main/resource/` and you are good to go. 

that said I would consider whether this feature is needed at all. I believe other clients don't offer loading a custom trust store. I originally introduced it, because I needed it for testing. But now there is the TLS INSECURE mode which does not validate certificates at all.

Reply:
in container world there is classloader hierarchy, it could be tricky to work out where to put trust store, because it is relative to `DelegatingTlsChannel.class`.

I agree on trust store, we should perhaps make it optional?

Reply:
I think in application containers a QuestDB library will be packaged as a part of a deployable application (WAR). So it's still `/src/main/resource`. It could be more complicated when QuestDB is deployed globally for all applications. 

truststore already *is* optional. If you don't specify it explicitly then it will default to a system-wide trustore. So when a server cert is signed by a trusted authority (e.g. QuestDB cloud) then you don't need to set a custom trust store at all. This is how it's exposed in the SenderBuilder:
```java
public class AdvancedTlsSettings {
            /**
             * Configure a custom truststore. This is only needed when using {@link #enableTls()} when your default
             * truststore does not contain certificate chain used by a server. Most users should not need it.
[...]
```